pkg/zrouter/zmiddlewares/metrics.go

```diff
@@ -0,0 +1,86 @@
+package zmiddlewares
+
+import (
+	"fmt"
+	"github.com/go-chi/chi/v5"
+	"github.com/zondax/golem/pkg/metrics"
+	"github.com/zondax/golem/pkg/metrics/collectors"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+const (
+	activeConnectionsMetricType    = "active_connections"
+	durationMillisecondsMetricType = "duration_milliseconds"
+	responseSizeMetricType         = "response_size_bytes"
+	totalRequestMetricType         = "total_requests"
+	pathLabel                      = "path"
+	methodLabel                    = "method"
+	statusLabel                    = "status"
+)
+
+func RegisterRequestMetrics(appName string, metricsServer metrics.TaskMetrics) []error {
+	var errs []error
+
+	register := func(name, help string, labels []string, handler metrics.MetricHandler) {
+		if err := metricsServer.RegisterMetric(name, help, labels, handler); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	totalRequestsMetricName := getMetricName(appName, totalRequestMetricType)
+	responseSizeMetricName := getMetricName(appName, responseSizeMetricType)
+	durationMillisecondsMetricName := getMetricName(appName, durationMillisecondsMetricType)
+	activeConnectionsMetricName := getMetricName(appName, activeConnectionsMetricType)
+	register(totalRequestsMetricName, "Total number of HTTP requests made.", []string{methodLabel, pathLabel, statusLabel}, &collectors.Counter{})
+	register(durationMillisecondsMetricName, "Duration of HTTP requests.", []string{methodLabel, pathLabel, statusLabel}, &collectors.Histogram{})
+	register(responseSizeMetricName, "Size of HTTP response in bytes.", []string{methodLabel, pathLabel, statusLabel}, &collectors.Histogram{})
+	register(activeConnectionsMetricName, "Number of active HTTP connections.", nil, &collectors.Gauge{})
+
+	if len(errs) > 0 {
+		return errs
+	}
+
+	return nil
+}
+
+func RequestMetrics(appName string, metricsServer metrics.TaskMetrics) Middleware {
+	var activeConnections int64
```

Comment:
I only have one question. Wouldn't this require a lock because of concurrent writes?

Reply:
Yes! Good catch